### PR TITLE
Docs: casting temporal fields to ordinal fields + 

### PR DESCRIPTION
**Pull request by kanitw**
also re-organize timeUnit page to have supported values up front.

Fix https://github.com/vega/vega-lite/issues/3131
also part of https://github.com/vega/vega-lite/issues/2279

### CODE REVIEW

File: build/vega-lite-schema.json [hunk 1]
```diff
@@ -1184,11 +1184,11 @@
         },
         "timeUnit": {
           "$ref": "#/definitions/TimeUnit",
-          "description": "Time unit for a `temporal` field  (e.g., `year`, `yearmonth`, `month`, `hour`).\n\n__Default value:__ `undefined` (None)"
+          "description": "Time unit (e.g., `year`, `yearmonth`, `month`, `hours`) for a temporal field.\nor [a temporal field that gets casted as ordinal](type.html#cast).\n\n__Default value:__ `undefined` (None)"
         },
         "type": {
           "$ref": "#/definitions/Type",
-          "description": "The encoded field's type of measurement. This can be either a full type\nname (`\"quantitative\"`, `\"temporal\"`, `\"ordinal\"`,  and `\"nominal\"`)."
+          "description": "The encoded field's type of measurement (`\"quantitative\"`, `\"temporal\"`, `\"ordinal\"`,  or `\"nominal\"`)."
```
Comment (domoritz) [inline build/vega-lite-schema.json]: double space before or

File: site/docs/encoding/type.md [hunk 2]
```diff
@@ -3,25 +3,32 @@ layout: docs
...
-### Data Type
+## Quantitative
+Quantitative data expresses some kind of quantity. Typically this is numerical data. For example `7.3`, `42.0`, `12.1`.
 
-If a field is specified, the channel definition **must** describe the encoded data's [type of measurement (level of measurement)](https://en.wikipedia.org/wiki/Level_of_measurement).
-The supported data types are:
+## Temporal
+Temporal data supports date-times and times. For example `2015-03-07 12:32:17`, `17:01`, `2015-03-16`.
+
+Note that when a `"temporal"` type is used for a field, Vega-Lite will treat it as a continuous field and thus will use a [`time` scale](scale.html#time) to map its data to visual values. For example, the following bar chart shows the dimension of the bars of a time scale.
```
Comment (domoritz) [inline site/docs/encoding/type.md]: "shows the dimension of the bars of a time scale" what?

File: site/docs/encoding/type.md [hunk 3]
```diff
@@ -3,25 +3,32 @@ layout: docs
...
+
+<span class="vl-example" data-name="bar_month_temporal"></span>
+
+### Casting a Temporal Field as an Ordinal Field
```
Comment (domoritz) [inline site/docs/encoding/type.md]: Casting is a bad term for this. Casting means to convert types but in our case we are treating data differently. I'd prefer to say it that way.

File: site/docs/encoding/type.md [hunk 4]
```diff
@@ -3,25 +3,32 @@ layout: docs
...
+To treat a date-time field as a discrete field, you can cast it be an `"ordinal"` field.
+This casting strategy can be useful for time units with low cardinality such as `"month"`.
 
-Temporal
-: Temporal data supports date-times and times. For example `2015-03-07 12:32:17`, `17:01`, `2015-03-16`.
+<span class="vl-example" data-name="bar_month"></span>
 
-Ordinal
-: Ordinal data represents ranked order (1st, 2nd, ...) by which the data can be sorted. However, as opposed to quantitative data, there is no notion of *relative degree of difference* between them. For illustration, a "size" variable might have the following values `small`, `medium`, `large`, `extra-large`. We know that medium is larger than small and same for extra-large larger than large. However, we cannot claim that compare the magnitude of difference, for example, between (1) small and medium and (2) medium and large.
+## Ordinal
```
Comment (domoritz) [inline site/docs/encoding/type.md]: "However, we cannot claim that compare the magnitude of " broken sentence

File: site/docs/encoding/type.md [hunk 5]
```diff
@@ -3,25 +3,32 @@ layout: docs
...
 
-Nominal
-: Nominal data, also known as categorical data, differentiates between values based only on their names or categories. For example, gender, nationality, music genre, names are all nominal data. Numbers maybe used to represent the variables but the number do not determine magnitude or ordering. For example, if a nominal variable contains three values 1, 2, and 3. We cannot claim that 1 is less than 2 nor 3.
+## Nominal
+Nominal data, also known as categorical data, differentiates between values based only on their names or categories. For example, gender, nationality, music genre, names are all nominal data. Numbers maybe used to represent the variables but the number do not determine magnitude or ordering. For example, if a nominal variable contains three values 1, 2, and 3. We cannot claim that 1 is less than 2 nor 3.
```
Comment (domoritz) [inline site/docs/encoding/type.md]: "music genre, names" -> "music genre, and name" (add and and remove plural)

File: site/docs/mark/bar.md [hunk 6]
```diff
@@ -16,8 +16,7 @@ permalink: /docs/bar.html
 }
 ```
 
-The `bar` mark encodes x and y channels with a pair of discrete and continuous fields. Bar marks are useful in a wide variety of visualizations, including bar charts and timelines.
-
+Bar marks are useful in a wide variety of visualizations, including bar charts, [stacked bar charts](#stack), and [timelines](#ranged).
```
Comment (domoritz) [inline site/docs/mark/bar.md]: in many visualizations

File: site/docs/mark/bar.md [hunk 7]
```diff
@@ -39,6 +38,16 @@ If we map a different discrete field to the `y` channel, we can produce a horizo
 
 <span class="vl-example" data-name="bar_aggregate"></span>
 
+While the `bar` mark typically encodes x and y channels with a pair of discrete and continuous fields, it can also be used with continuous fields on both x and y channels.
```
Comment (domoritz) [inline site/docs/mark/bar.md]: uses the x and y channels to encode a pair of discrete and continuous fields (reverse the direction of encode)

File: site/docs/mark/bar.md [hunk 8]
```diff
@@ -39,6 +38,16 @@ If we map a different discrete field to the `y` channel, we can produce a horizo
...
+For example, given a bar chart with a temporal field on x, we can see that the x-scale is a continuous scale. By default, the size of bars on continuous scales will be set based on the [`continuousBandSize` config](#config).
+
+<span class="vl-example" data-name="bar_month_temporal"></span>
+
+If you want to use a discrete scale instead, you can cast the field to have an `"ordinal"` type. This casting strategy can be useful for time units with low cardinality such as `"month"`.
```
Comment (domoritz) [inline site/docs/mark/bar.md]: rename case

File: site/docs/transform/timeunit.md [hunk 9]
```diff
@@ -35,7 +49,15 @@ Time unit is used to discretize times in Vega-Lite. It can either be used [insid
...
-The chart below shows the average precipitation in Seattle over multiple years aggregated by month.
+A field definition can include a `timeUnit` property.  For example, the chart below shows shows temperature in Seattle aggregated by month.
+
+<span class="vl-example" data-name="line_month"></span>
+
+Note that temporal fields use continuous scales by default for all mark types including `"bar"`.
+
+<span class="vl-example" data-name="bar_month_temporal"></span>
+
+If you want to use a discrete scale instead, you can cast the field to have an `"ordinal"` type. This casting strategy can be useful for time units with low cardinality such as `"month"`.
```
Comment (domoritz) [inline site/docs/transform/timeunit.md]: replace cast

File: site/docs/encoding/type.md (hunk 3 above)
Comment (kanitw) [reply site/docs/encoding/type.md]: I disagree.  When I talk to Jeff, he also that this casting seems natural.

**Comment by kanitw** [comment]
I'll be on a flight. You can fix these so we can merge.  (After all, some of these changes are transforms docs which suppose to be on your plate anyway.)

File: site/docs/transform/timeunit.md (hunk 9 above)
Comment (kanitw) [reply site/docs/transform/timeunit.md]: Disagree

File: site/docs/mark/bar.md (hunk 8 above)
Comment (kanitw) [reply site/docs/mark/bar.md]: Disagree

File: site/docs/encoding/type.md (hunk 3 above)
Comment (domoritz) [reply site/docs/encoding/type.md]: Okay

**Pull request merged**